lib/queryBuilder/graphUpserter/UpsertNode.js

```diff
@@ -96,6 +97,14 @@ function hasChanges(currentModel, upsertModel, queryProps) {
 
     if (!relations[key] && currentModel[key] !== upsertModel[key]) {
       return true;
+    } else if (relations[key] && relations[key].constructor.name === BelongsToOneRelation.name) {
+      const idColumn = currentModel.constructor.idColumn;
```

Comment:
You can use `relations[key] instanceof BelongsToOneRelation` instead of comparing constructor names.

Reply:
👍